src/macaron/repo_finder/repo_finder.py

```diff
@@ -133,7 +133,7 @@ def to_repo_path(purl: PackageURL, available_domains: list[str]) -> str | None:
     )
 
 
-def find_source(purl_string: str, input_repo: str | None) -> bool:
+def find_source(purl_string: str, input_repo: str | None, latest_version_fallback: bool = True) -> bool:
```

Comment:
It seems that the repo finder used in the [analyzer](https://github.com/oracle/macaron/blob/a2b6ba929778f97e3cb6ac1453230e78a89fbce4/src/macaron/slsa_analyzer/analyzer.py#L783) module is inconsistent with this implementation, as it doesn't utilize the same fallback for the latest version?

Reply:
It does use the same mechanism just through a different path. The repository is potentially re-found during the [prepare_repo](https://github.com/oracle/macaron/blob/a2b6ba929778f97e3cb6ac1453230e78a89fbce4/src/macaron/repo_finder/repo_utils.py#L129) function call in `repo_utils.py`